eva/readers/decord_reader.py

```diff
@@ -108,12 +121,34 @@ def _read(self) -> Iterator[Dict]:
                 if begin % self._sampling_rate:
                     begin += self._sampling_rate - (begin % self._sampling_rate)
                 for frame_id in range(begin, end + 1, self._sampling_rate):
-                    frame = video[frame_id].asnumpy()
+                    frame = self.__get_frame(frame_id, av_reader, v_reader)
                     if frame is not None:
-                        yield {
-                            "id": frame_id,
-                            "data": frame,
-                            "seconds": round(video.get_frame_timestamp(frame_id)[0], 2),
-                        }
+                        yield frame
                     else:
                         break
+
+    def __get_frame(self, frame_id, av_reader, v_reader):
```

Comment:
We shouldn't hit these decisions for every frame. How about we have 3 functions and use logic upfront to decide the correct callable and then just invoke it?